File: aoc2018/aoc2018-java/src/test/java/de/havox_design/aoc2018/day07/Day07Test.java
package de.havox_design.aoc2018.day07;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day07Test {

    @ParameterizedTest
    @MethodSource("getDataForTask1")
    void testTask1(String fileName, String expectation) {
        Assertions.assertEquals(expectation, Day07.processTask1(fileName));
    }

    private static Stream<Arguments> getDataForTask1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2018/day07/day07part1sample.txt", "CABDFE")
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForTask2")
    void testTask2(String fileName, long expectation) {
        Assertions.assertEquals(expectation, Day07.processTask2(fileName));
    }

    private static Stream<Arguments> getDataForTask2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2018/day07/day07part2sample.txt", 0L)
        );
    }
}

File: aoc2015/day07/src/test/kotlin/de/havox_design/aoc2015/day07/Day07Test.kt
package de.havox_design.aoc2015.day07

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day07Test {
    @Test
    fun testMainClass() {
        MainClass.main(arrayOf())
    }

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart1")
    fun testProcessPart1(filename: String, expectedResult: Int) =
        LogicGates(filename).processPart1().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestProcessPart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("day07Part1sample1.txt", 72),
                Arguments.of("day07Part1sample2.txt", 507),
                Arguments.of("day07Part1sample3.txt", 492),
                Arguments.of("day07Part1sample4.txt", 114),
                Arguments.of("day07Part1sample5.txt", 65412),
                Arguments.of("day07Part1sample6.txt", 65079),
                Arguments.of("day07Part1sample7.txt", 123),
                Arguments.of("day07Part1sample8.txt", 456)
            )
    }
}

private fun Int.shouldBe(expectation: Int) = Assertions.assertEquals(expectation, this)
